include/wil/cppwinrt_helpers.h

```diff
@@ -212,6 +212,48 @@ namespace wil::details
 #endif // __WIL_CPPWINRT_MICROSOFT_UI_DISPATCHING_HELPERS
 /// @endcond
 
+#if defined(WINRT_Windows_Foundation_H) && !defined(__WIL_CPPWINRT_WINDOWS_FOUNDATION_HELPERS)
+#define __WIL_CPPWINRT_WINDOWS_FOUNDATION_HELPERS
+namespace Windows::Foundation
+{
+    /// @cond
+    struct IMemoryBufferByteAccess;
+    /// @endcond
+}
+
+namespace wil
+{
+    //! Returns a view into the underlying bytes of a memory buffer
+    //! provided in the form of an IMemoryBufferReference.
+    //! The caller is responsible for ensuring that the memory buffer's
+    //! lifetime encompasses the lifetime of the returned view.
+    //! By default, returns an array_view<uint8_t>, but you can provide an alternate
+    //! type such as to_array_view<double>.
+    //! You must include memorybuffer.h in order to use this overload of to_array_view.

```

Comment:
Unfortunately, we have to wait for the client to `#include <memorybuffer.h>` to get access to `IMemoryBufferByteAccess`. C++/WinRT avoids this requirement by manually declaring the interface. Should we do that here?

Reply:
Any concern to us just including the necessary header? Or can we add its header to the ifdef?

Reply:
Unfortunately, `memorybuffer.h` relies on `#pragma once` and does not have a `#define` for detecting whether it has been included.

I'm okay with explicitly including the header if you think it's sufficiently unobtrusive.

Reply:
Hmm, looks like the only thing it includes is `objidl.h`. While I'd like to believe that it is innocuous, I'm not so sure of that. I'd be okay defining it ourselves _but_ wouldn't we have the same issue with `IUnknown` as well?

Reply:
Ultimately, I think this is probably fine. The alternative would be to move the definition to a new header that does the necessary inclusion that the user would then need to explicitly include, so not _that_ much different than needing to include an additional header. We may want to consider adding a comment above the line where the compilation error will be produced saying something like "if you receive an error saying that this type is not defined, include this header."

Reply:
I'll require the caller to include for now. We can relax that later as a non-breaking change.